### PR TITLE
docs: update .example.env and README

### PR DESCRIPTION
**Pull request by ciaransweet**
# What this PR is

This PR is just me updating the `.example.env` file to include variables I had to add when deploying `veda-backend` from scratch. I left them empty as I assume they're required as it fell over without them. I removed 2 variables as they weren't used across the codebase too.

I also updated the `README` to include the `ecr` login step, I guess most have done this previously but on a fresh machine, it was a requirement!

### CODE REVIEW

**Review by smohiudd — APPROVED** [review]

File: .example.env [hunk 1]
```diff
@@ -27,6 +24,12 @@ VEDA_DB_PUBLICLY_ACCESSIBLE=TRUE
 VEDA_RASTER_ROOT_PATH=
 VEDA_STAC_ROOT_PATH=
 
+VEDA_USERPOOL_ID=
+VEDA_CLIENT_ID=
+VEDA_CLIENT_SECRET=secret
```
Comment (botanical) [inline .example.env:29]: Where is `VEDA_CLIENT_SECRET` used? 🤔

Comment (botanical) [reply .example.env:29]: Ah, I think I found it https://github.com/NASA-IMPACT/veda-backend/blob/develop/ingest_api/infrastructure/config.py#L43!

File: .example.env [hunk 2]
```diff
@@ -27,6 +24,12 @@ VEDA_DB_PUBLICLY_ACCESSIBLE=TRUE
...
+VEDA_DATA_ACCESS_ROLE_ARN=
+VEDA_COGNITO_DOMAIN=
+
```
Comment (botanical) [inline .example.env:32]: While you're modifying this file, can you also add:
- VEDA_DB_USE_RDS_PROXY=[OPTIONAL]
- VEDA_DB_RDS_INSTANCE_CLASS=[OPTIONAL]
- VEDA_DB_RDS_INSTANCE_SIZE=[OPTIONAL]

?

**Review by botanical — APPROVED** [review]
Thanks for updating the readme and .example.env!

**Review by anayeaye — APPROVED** [review]

File: .example.env (hunk 2 above)
Comment (ciaransweet) [reply .example.env:34]: Done

**Pull request merged**